Gosu/Audio.hpp

```diff
@@ -33,33 +38,38 @@ namespace Gosu
         //! Calling this twice, or too late, does not do any harm.
         void stop();
 
+        //! Returns the current volume of the song.
+        double volume() const;
         //! \param volume Can be anything from 0.0 (silence) to 1.0 (full volume).
         void set_volume(double volume);
+
         //! \param pan Can be anything from -1.0 (left) to 1.0 (right).
         void set_pan(double pan);
         //! \param speed Use 1.0 for normal playback speed.
         void set_speed(double speed);
     };
-    
-    //! A sample is a short sound that is completely loaded in memory, can be
-    //! played multiple times at once and offers very flexible playback
-    //! parameters. Use samples for everything that's not music.
-    class Sample
-    {
-        struct SampleData;
-        std::shared_ptr<SampleData> data;
 
+
+    //! A sound handles all your audio needs from short samples (e.g. explosions) to long songs (e.g. background music)
+    //!
+    //! Use streaming = false for everything that's not music. (Maybe add a more detailed explanation why this is better,
+    //! because some the disadvantages of song might dissappear with the merge?)
+    class Sound
+    {
+        struct SoundData;
+        std::shared_ptr<SoundData> data;
+        std::vector<Channel> current_songs;
     public:
         //! Constructs an empty sample that acts as if the song had a length of 0.
-        Sample();
+        Sound();
         
         //! Constructs a sample that can be played on the specified audio
         //! system and loads the sample from a file.
-        explicit Sample(const std::string& filename);
+        explicit Sound(const std::string& filename, bool streaming = false);
```

Comment:
Are there named parameters in c++. Googling resulted in different opionions on how to and if at all...

Reply:
Not really. You can build them yourself but I try to use only basic features in Gosu (both C++ and Ruby).